modules/pricer/src/main/java/com/opengamma/strata/pricer/ZeroRatePeriodicDiscountFactors.java

```diff
@@ -194,14 +200,18 @@ public double discountFactorTimeDerivative(double yearFraction) {
 
   @Override
   public double zeroRate(double yearFraction) {
-    double ratePeriod = curve.yValue(yearFraction);
+    double yearFractionMod = Math.max(EFFECTIVE_ZERO, yearFraction);
+    double ratePeriod = curve.yValue(yearFractionMod);
```

Comment:
Is this consitent with `discountFactor` and `zeroRatePointSensitivity`?

Reply:
I have replaced the above and done the same in ZeroRateDiscountFactors.java to ensure we always return 0 if year faction is not greater than 0.